hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/ClientConnectionManagerImpl.java

```diff
@@ -78,9 +78,16 @@
 import static com.hazelcast.client.spi.properties.ClientProperty.HEARTBEAT_INTERVAL;
 import static com.hazelcast.client.spi.properties.ClientProperty.HEARTBEAT_TIMEOUT;
 
+/**
+ * Implementation of {@link ClientConnectionManager}.
+ */
 public class ClientConnectionManagerImpl implements ClientConnectionManager {
 
-    private final NonBlockingIOThreadOutOfMemoryHandler OUT_OF_MEMORY_HANDLER = new NonBlockingIOThreadOutOfMemoryHandler() {
+    protected final AtomicInteger connectionIdGen = new AtomicInteger();
+
+    protected volatile boolean alive;
+
+    private final NonBlockingIOThreadOutOfMemoryHandler outOfMemoryHandler = new NonBlockingIOThreadOutOfMemoryHandler() {
```

Comment:
I tried that first, but the handler uses another instance field (the logger instance), so it cannot be `static`. I didn't want to mess around with the implementation, so I just fixed the name for now.